### ISSUE TITLE
Cache for specific pipeline

### ISSUE THREAD

**Issue by joepjoosten** [wercker/support#173]
When pushing docker images (internal/docker-push) during the deploy pipeline, i noticed that the cache is in the pushed image.

This makes the pushed image unnecessary large.

Deleting it before pushing will kill the cache, and will increase build times

There should be a way to accomplish a small docker image, and not using docker-scratch-push. (Not everyone is using go... ;)

Possible solution:
Have to different caches
1. For all pipelines (dev/build/deploy): can be the current one: $WERCKER_CACHE_DIR
2. Per pipeline cache: e.g. $WERCKER_PIPELINE_CACHE_DIR

This way build specific cache can be in the $WERCKER_PIPELINE_CACHE_DIR, and won't be in the pushed docker image.

**Comment by Tonnu** [comment]
Thanks for the suggestion :). I'll create an internal ticket and discuss it with the team.

**Comment by ernoaapa** [comment]
Any news about this? @Tonnu clearing $WERCKER_CACHE_DIR doesn't work as @joepjoosten said, it would kill the caching.

**Comment by joepjoosten** [comment]
Other solution: somehow exclude the cache in the docker-push image...

**Comment by ernoaapa** [comment]
Yeah, I tried to have `.dockerignore` in the root of the project but didn't work. 
I tried following `.dockerignore` file:
```
/pipeline/cache
pipeline/cache
cache
/cache
.cache
../cache
/pipeline/.cache
pipeline/.cache
```

And copied it to:
- `/pipeline/source`
- `/pipeline`
- `/`

But didn't work.

If the `internal/docker-push` source code would be somewhere, I could find some workaround, but now it's quite challenging because don't know how it builds the docker image. Also would not like to switch to use some 3rd party docker step because most of them are really outdated.
